projects/reservoir/fv3/era_process.py

```diff
@@ -37,18 +62,34 @@ def main(args):
     setup_fregrid()
     for var in variables:
         check_time_steps_complete(path, var)
-        weekdays = ["mon", "tue", "wed", "thu", "fri", "sat", "sun"]
-        for i, day in enumerate(weekdays):
-            merge_into_file(path, var, i, day)
-            logger.info(f"{day} is merged")
-            regrid_to_cubed_sphere(path, var, day)
-            logger.info(f"{day} is regridded")
-            interpolated = interpolate_nans(var, day)
-            logger.info(f"{day} is interpolated")
-            masked = mask(interpolated)
-            logger.info(f"{day} is masked")
-            masked.to_netcdf(args.out_path + var + "_" + day + ".nc")
-            logger.info(f"{day} is done")
+        merge_into_weekly_file(path, var, args)
```

Comment:
Instead of calling the sequence of regridding and masking for each variable, can they all just be merged at the start? e.g. something like
```
for var in variables:
    check_time_steps_complete(...)
full_variables = merge_variables(path, variables)
# do time shift, add tendency by doing xarray operations on the full dataset, no need to write it in between steps
full_variables["sst_tendency"] = calculate_sst_tendency(...)
for var in TIME_SHIFT_VARIABLES:
    full_variables[var] = full_variables[var].shift(time=-1)
# masking, etc.
...
```

Reply:
Maybe, but I was worried that fregrid is going to cause problems

Reply:
Okay, that#s done. It worked